scipy/signal/signaltools.py

```diff
@@ -2338,6 +2338,13 @@ def resample_poly(x, up, down, axis=0, window=('kaiser', 5.0)):
     window : string, tuple, or array_like, optional
         Desired window to use to design the low-pass filter, or the FIR filter
         coefficients to employ. See below for details.
+    border : string, optional
+        `zero`, `mean` or `line`. Changes assumptions on the values beyond the
+        boundary. If `zero`, assumed to be zero. If `line` assumed to continue
+        a linear trend defined by the first and last points. `mean`, `median`,
+        `maximum` and `minimum` work as in np.pad and assume that the values
+        beyond the boundary are the mean, median, maximum or minimum
+        respectively of the array along the axis.
```

Comment:
Could we get the functionality you need by supporting a subset of the options in `padlen` for `filtfilt`?

Hopefully whatever we do here will be forward compatible with @grlee77's code.

Reply:
The implementation here does not support any options for padlen of filtfilt

If you mean padtype, I can support constant by creating another option called cval, as in @grleee77 PR #10552 or constant_values as in np.pad (but it is misleading, as we use a single axis and thus a single constant). Supporting 'even' and 'constant' are not possible with the implementation in this PR.

Reply:
I used cval like in #10552 at the end